### PR TITLE
@LoadBalanced RestClient

### PR DESCRIPTION
**Pull request by OlgaMaciaszek**
Fixes gh-1293.

### CODE REVIEW

**Comment by OlgaMaciaszek** [comment]
Hi @Buzzardo, could you please review documentation changes?

**Review by ryanjbaxter — APPROVED** [review]

File: docs/modules/ROOT/pages/spring-cloud-commons/common-abstractions.adoc [hunk 1]
```diff
@@ -135,7 +135,7 @@ Please see the documentation of the `ServiceRegistry` implementation you use for
 For instance, Eureka's supported statuses are `UP`, `DOWN`, `OUT_OF_SERVICE`, and `UNKNOWN`.
 
 [[rest-template-loadbalancer-client]]
-== Spring RestTemplate as a Load Balancer Client
+== Spring `RestTemplate` as a LoadBalancer Client
```
Comment (Buzzardo) [inline docs/modules/ROOT/pages/spring-cloud-commons/common-abstractions.adoc:138]: Remove the ticks around `RestTemplate`. Your instinct is right, but we have had trouble with rendering code fonts in headings, so we had to remove it.

File: docs/modules/ROOT/pages/spring-cloud-commons/common-abstractions.adoc [hunk 2]
```diff
@@ -167,13 +167,140 @@ CAUTION: A `RestTemplate` bean is no longer created through auto-configuration.
...
 The URI needs to use a virtual host name (that is, a service name, not a host name).
-The BlockingLoadBalancerClient is used to create a full physical address.
+The `BlockingLoadBalancerClient` is used to create a full physical address.
 
-IMPORTANT: To use a load-balanced `RestTemplate`, you need to have a load-balancer implementation in your classpath.
+IMPORTANT: To use a load-balanced `RestTemplate`, you need to have a Spring Cloud LoadBalancer implementation in your classpath.
 Add xref:spring-cloud-commons/loadbalancer.adoc#spring-cloud-loadbalancer-starter[Spring Cloud LoadBalancer starter] to your project in order to use it.
 
+[[multiple-resttemplate-objects]]
+=== Multiple `RestTemplate` Objects
```
Comment (Buzzardo) [inline docs/modules/ROOT/pages/spring-cloud-commons/common-abstractions.adoc:176]: Remove the ticks around `RestTemplate`. As I mentioned in the previous comment, we have trouble with rendering the code font in headings.

File: docs/modules/ROOT/pages/spring-cloud-commons/common-abstractions.adoc [hunk 3]
```diff
@@ -167,13 +167,140 @@ CAUTION: A `RestTemplate` bean is no longer created through auto-configuration.
...
+    }
+}
+----
+
+IMPORTANT: Notice the use of the `@Primary` annotation on the plain `RestTemplate` declaration in the preceding example to disambiguate the unqualified `@Autowired` injection.
+
+TIP: If you see errors such as `java.lang.IllegalArgumentException: Can not set org.springframework.web.client.RestTemplate field com.my.app.Foo.restTemplate to com.sun.proxy.$Proxy89`, try injecting `RestOperations` or setting `spring.aop.proxyTargetClass=true`.
+
+[[rest-client-loadbalancer-client]]
+== Spring `RestClient` as a LoadBalancer Client
```
Comment (Buzzardo) [inline docs/modules/ROOT/pages/spring-cloud-commons/common-abstractions.adoc:222]: Remove the ticks around `RestClient`. It won't render correctly.

File: docs/modules/ROOT/pages/spring-cloud-commons/common-abstractions.adoc [hunk 4]
```diff
@@ -167,13 +167,140 @@ CAUTION: A `RestTemplate` bean is no longer created through auto-configuration.
...
+        return restClientBuilder.build().get().uri(URI.create("http://stores/stores")).retrieve().body(String.class);
+    }
+}
+----
+
+The URI needs to use a virtual host name (that is, a service name, not a host name).
+The `BlockingLoadBalancerClient` is used to create a full physical address.
+
+IMPORTANT: To use a load-balanced `RestClient`, you need to have a Spring LoadBalancer implementation in your classpath.
+Add xref:spring-cloud-commons/loadbalancer.adoc#spring-cloud-loadbalancer-starter[Spring Cloud LoadBalancer starter] to your project in order to use it.
```
Comment (Buzzardo) [inline docs/modules/ROOT/pages/spring-cloud-commons/common-abstractions.adoc:254]: Change to:

To use it, add xref:spring-cloud-commons/loadbalancer.adoc#spring-cloud-loadbalancer-starter[Spring Cloud LoadBalancer starter] to your project.

File: docs/modules/ROOT/pages/spring-cloud-commons/common-abstractions.adoc [hunk 5]
```diff
@@ -167,13 +167,140 @@ CAUTION: A `RestTemplate` bean is no longer created through auto-configuration.
...
+        				.retrieve().body(String.class);
+    }
+}
+----
+
+IMPORTANT: Notice the use of the `@Primary` annotation on the plain `RestTemplate` declaration in the preceding example to disambiguate the unqualified `@Autowired` injection.
+
 [[webclinet-loadbalancer-client]]
-== Spring WebClient as a Load Balancer Client
+== Spring `WebClient` as a LoadBalancer Client
```
Comment (Buzzardo) [inline docs/modules/ROOT/pages/spring-cloud-commons/common-abstractions.adoc:303]: Remove the ticks around `WebClient`. The code font does not render correctly in headings.

File: docs/modules/ROOT/pages/spring-cloud-commons/common-abstractions.adoc [hunk 6]
```diff
@@ -204,11 +331,55 @@ public class MyClass {
...
 The Spring Cloud LoadBalancer is used to create a full physical address.
 
-IMPORTANT: If you want to use a `@LoadBalanced WebClient.Builder`, you need to have a load balancer
+IMPORTANT: If you want to use a `@LoadBalanced WebClient.Builder`, you need to have a Spring Cloud LoadBalancer
 implementation in the classpath. We recommend that you add the
 xref:spring-cloud-commons/loadbalancer.adoc#spring-cloud-loadbalancer-starter[Spring Cloud LoadBalancer starter] to your project.
 Then, `ReactiveLoadBalancer` is used underneath.
 
+[[multiple-webclient-objects]]
+=== Multiple `WebClient.Builder` Objects
```
Comment (Buzzardo) [inline docs/modules/ROOT/pages/spring-cloud-commons/common-abstractions.adoc:340]: Remove the ticks around `WebClient.Builder`. The code font does not render correctly in headings.

File: docs/modules/ROOT/pages/spring-cloud-commons/loadbalancer.adoc [hunk 7]
```diff
@@ -44,12 +44,13 @@ NOTE: The classes you pass as `@LoadBalancerClient` or `@LoadBalancerClients` co
 [[spring-cloud-loadbalancer-integrations]]
 == Spring Cloud LoadBalancer integrations
 
-In order to make it easy to use Spring Cloud LoadBalancer, we provide `ReactorLoadBalancerExchangeFilterFunction` that can be used with `WebClient` and `BlockingLoadBalancerClient` that works with `RestTemplate`.
+In order to make it easy to use Spring Cloud LoadBalancer, we provide `ReactorLoadBalancerExchangeFilterFunction` that can be used with `WebClient` and `BlockingLoadBalancerClient` that works with `RestTemplate` and `RestClient`.
```
Comment (Buzzardo) [inline docs/modules/ROOT/pages/spring-cloud-commons/loadbalancer.adoc:47]: Change to:

To make it easy to use Spring Cloud LoadBalancer, we provide `ReactorLoadBalancerExchangeFilterFunction` (which can be used with `WebClient`) and `BlockingLoadBalancerClient` (which works with `RestTemplate` and `RestClient`).

--------

The parentheses break up a long sentence and make it more clear. "In order to" can nearly always be changed to "To". Edit to add: That's not your fault. That sentence had problems in the first place. While we have the chance, though, we may as well improve it.

File: docs/modules/ROOT/pages/spring-cloud-commons/loadbalancer.adoc [hunk 8]
```diff
@@ -232,7 +233,7 @@ public class CustomLoadBalancerConfiguration {
 ----
 
 TIP: For the non-reactive stack, create this supplier with the `withBlockingHealthChecks()`.
-You can also pass your own `WebClient` or `RestTemplate` instance to be used for the checks.
+You can also pass your own `WebClient`, `RestTemplate` or `RestClient` instance to be used for the checks.
```
Comment (Buzzardo) [inline docs/modules/ROOT/pages/spring-cloud-commons/loadbalancer.adoc:236]: Add a comma after `RestTemplate. The Spring Style Guide calls for serial commas (a comma before the last item in a list).

**Review by Buzzardo — CHANGES_REQUESTED** [review]
Good changes. Sorry about the code font not rendering in headings. Your instinct to put it in is correct, but it doesn't work well across our various outputs.

**Comment by OlgaMaciaszek** [comment]
Thanks, @Buzzardo 🙂

**Pull request merged**